src/geometry/lane_geometry.cc

```diff
@@ -127,26 +135,27 @@ maliput::api::RBounds LaneGeometry::RBounds(double p) const {
   MALIPUT_THROW_UNLESS(p >= p0());
   MALIPUT_THROW_UNLESS(p <= p1());
 
-  // Get p equivalent left and right.
   // Get distance from centerline to left and right.
-  const double p_left = FromCenterPToLateralP(LineStringType::kLeftBoundary, p);
-  const double p_right = FromCenterPToLateralP(LineStringType::kRightBoundary, p);
-
+  const maliput::math::Vector3 on_left = ToLateralPos(LineStringType::kLeftBoundary, p);
+  const maliput::math::Vector3 on_right = ToLateralPos(LineStringType::kRightBoundary, p);
   const maliput::math::Vector3 on_centerline = utility::InterpolatedPointAtP(centerline_, p);
-  const maliput::math::Vector3 on_left = utility::InterpolatedPointAtP(left_, p_left);
-  const maliput::math::Vector3 on_right = utility::InterpolatedPointAtP(right_, p_right);
   return {-(on_right - on_centerline).norm(), (on_left - on_centerline).norm()};
 }
 
 double LaneGeometry::FromCenterPToLateralP(const LineStringType& line_string_type, double p) const {
   MALIPUT_THROW_UNLESS(line_string_type != LineStringType::kCenterLine);
-  // For computing the lane bounds the equivalent point in the lateral line string is needed.
   // Given p in centerline, let's compute p_at_a_border as: p_at_a_border = p / length * length_at_a_border.
   // See https://github.com/maliput/maliput_sparse/issues/15 for more details.
   const double lateral_length = line_string_type == LineStringType::kLeftBoundary ? left_.length() : right_.length();
   const double p_equivalent = p * lateral_length / centerline_.length();
   return std::clamp(p_equivalent, 0., lateral_length);
 }
 
+maliput::math::Vector3 LaneGeometry::ToLateralPos(const LineStringType& line_string_type, double p) const {
+  MALIPUT_THROW_UNLESS(line_string_type != LineStringType::kCenterLine);
+  const double p_lateral = FromCenterPToLateralP(line_string_type, p);
+  return utility::InterpolatedPointAtP(line_string_type == LineStringType::kLeftBoundary ? left_ : right_, p_lateral);
```

Comment:
No because of the same that here https://github.com/maliput/maliput_sparse/pull/38#discussion_r1003255820

I also tried to replace the enum for constexpr variables but given the parameter used as condition comes from the parameter AND the method isn't constexpr we can't solve that in build time.